### PR TITLE
feat: added file upload

### PR DESCRIPTION
**Pull request by fominmaksim**

### CODE REVIEW

**Review by alexandrtovmach — APPROVED** [review]

File: src/repositories/user.repository.ts [hunk 1]
```diff
@@ -46,10 +50,26 @@ export const updateOne = async (id: string, data: User) => {
       }
     );
   } else {
-    await UserModel.updateOne(
-      { _id: id },
-      { ...data, updatedAt: new Date().toISOString() }
-    );
+    const base64Image = file?.buffer.toString("base64");
+
+    if (base64Image) {
```
Comment (VitaliiDenysiuk) [inline src/repositories/user.repository.ts]: wrong place.
if somebody send data with password and file: first block if..else will be work and all your logic will not work. Such way you will write at database all info from field avatar without any logic

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/repositories/user.repository.ts (hunk 1 above)
Comment (alexandrtovmach) [reply src/repositories/user.repository.ts]: yeah, agree, that what I tried to understand on live review, but not get it

Comment (alexandrtovmach) [reply src/repositories/user.repository.ts]: I believe that mongoose ignore field if it's undefined, and not updating it, so this code block can be refactored to:
```ts
await UserModel.updateOne(
  { _id: id },
  {
    ...data,
    updatedAt: new Date().toISOString(),
    password: data.password && hashPassword(data.password),
    profile: { avatar: file && file?.buffer.toString("base64") },
  }
);
```

@fominmaksim could you test this approach, including password change and avatar, and see if it works. Thanks

**Review by VitaliiDenysiuk — APPROVED** [review]
lgtm

**Pull request merged**